src/dotnet/ToolManifest/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+// TODO Remove once we have https://github.com/dotnet/corefx/issues/34894
+
+namespace Microsoft.DotNet.Cli.Utils
```

Comment:
Per https://github.com/dotnet/corefx/issues/34894
to check in this type in corefx need more time. But I I do want to get more feed back on the behavior side in preview 4 so I checkin the type to unblock this work